src/cmd/build.js

```diff
@@ -30,31 +33,48 @@ export default function build(
     fileFilter = new FileFilter();
   }
 
-  return resolveManifest
-    .then((manifestData) =>
-      Promise.all([
-        prepareArtifactsDir(artifactsDir),
-        zipDir(sourceDir, {
-          filter: (...args) => fileFilter.wantFile(...args),
-        }),
-      ])
-      .then((results) => {
-        let [artifactsDir, buffer] = results;
-        let packageName = safeFileName(
-          `${manifestData.name}-${manifestData.version}.xpi`);
-        let extensionPath = path.join(artifactsDir, packageName);
-        let stream = createWriteStream(extensionPath);
-        let promisedStream = streamToPromise(stream);
-
-        stream.write(buffer, () => stream.end());
-
-        return promisedStream
-          .then(() => {
-            log.info(`Your web extension is ready: ${extensionPath}`);
-            return {extensionPath};
-          });
-      })
-    );
+  if (!createPackage) {
+    createPackage = () => {
```

Comment:
The `createPackage` parameter is probably needed to simplify the testing of this feature, how about refactoring this `createPackage` helper out of the `build` function body (mostly for readability reasons), e.g. I imagine that it could become something like:

```js
const defaultCreatePackage = (
  {resolveManifest, fileFilter, sourceDir, artifactsDir}: Object): Function => {
  return function() {
     // the same body of the original arrow function
  };
};

export default function build(
  ...
  if (!createPackage) {
    createPackage = defaultCreatePackage({
      resolveManifest,
      fileFilter,
      sourceDir,
      artifactsDir,
    });
  }
  ...
```